Replace Contact Summary Events tab with Admin UI Search Display and Form

guyiac opened the pull request:
Overview
----------------------------------------
Replace Contact Summary Events tab with Admin UI Search Display and Form per Coleman's request


Before
----------------------------------------
Standard Events Tab

After
----------------------------------------
Events tab of Contact Summary is a Search Display in a FormBuilder Form

Technical Details
----------------------------------------
Pretty straightforward

Comments
----------------------------------------
First commit ever. Be kind.

guyiac committed:
Replace Contact Summary Events tab with Admin UI Search Display and Form

civicrm-builder commented:
Can one of the admins verify this patch?

civicrm-builder commented:
Can one of the admins verify this patch?

seamuslee001 commented:
Jenkins add to whitelist

kcristiano commented:
jenkins test this please

colemanw commented on ext/civicrm_admin_ui/ang/afsearchEvents.aff.html:
```diff
@@ -0,0 +1,9 @@
+  <div class="af-markup">
+    <div class="help">
+    <p>This page lists all event registrations for this contact since inception. Click Add Event Registration below to register this contact for an event. Click Submit Credit Card Event Registration to process a new registration on behalf of the participant using their credit card.</p>
+    </div>
```
We need to add `ts()` around this test. I started doing that, and that got me thinking about the text itself and how it's a big mouthful of nothing. It just explains a self-explanatory table, then tells you you can click on some self-explanatory buttons and that they will do the thing they already say they will do.

colemanw committed:
Add ParticipantLinksProvider

Checks permissions and conditionally adds a credit card registration link

colemanw commented:
Ok I've reviewed, tested & cleaned this up, using a linksProvider callback to give the multiple register + credit card buttons.
The one thing I removed is the self-service "Transfer or Cancel" link which seems more appropriate to the front-end than the backend. I see it was added in #10695 but I still don't understand the use-case for it. It's still accessible by clicking the "View" link and then the self-service button is there... maybe that's good enough for whoever wanted it added?

guyiac commented:
> Ok I've reviewed, tested & cleaned this up, using a linksProvider callback to give the multiple register + credit card buttons. The one thing I removed is the self-service "Transfer or Cancel" link which seems more appropriate to the front-end than the backend. I see it was added in #10695 but I still don't understand the use-case for it. It's still accessible by clicking the "View" link and then the self-service button is there... maybe that's good enough for whoever wanted it added?

Isn't there a use case for an admin transferring or cancelling a registration if someone calls or shows up at the front desk?

eileenmcnaughton commented:
@colemanw this has passed & is MOP but just want to check that you have seen @guyiac's comment

colemanw committed:
AdminUI - Add self-service event cancel/transfer link to backend form

I don't know what purpose this serves, but it was added in https://github.com/civicrm/civicrm-core/pull/10695
so this keeps the new screen true to the old one.

colemanw commented:
@guyiac @eileenmcnaughton ok I have added the self-serve link back in.

francescbassas commented:
This introduces a little regression. Someone can review it? https://github.com/civicrm/civicrm-core/pull/30086
